### ISSUE TITLE
The machine gun MX LSW Jammed is too often in the prone (automatic fire).

### ISSUE THREAD

**Issue by ZeusWrath** [acemod/ACE3#4915]
**Arma 3 Version:** `1.66` (stable)
**CBA Version:** `3.1.2.161105` (stable)
**ACE3 Version:** `3.9.0` (stable)

**Mods:**
- `@CBA_A3`
- `@ace`

**Description:**
If a player plays for NATO autorifleman and shoot in automatic mode, then, less than a minute, the machinegun MX LSW jammed! I spent several tests. Sometimes the MX LSW can be jammed at the 16-18 seconds of fire of the first magazine! Just a less one minute and your weapon is not working! Sorry, but it looks like great nonsense. This feature can't be in real life, because machinegun are designed specifically for dense suppressive fire of long bursts. 
I think, the jamming probability  of the all machine guns in the ACE3, should be reduced and minimized. As seems to me, the minimal threshold for the jamming probability should be > 5 min.  If a player shoots for more than 5 minutes, only after this, to be the probability of jamming, in addition, it must be negligible.

**Steps to reproduce:**
 Place in the airfield of Stratis(clean and flat surface), the autorifleman (NATO), go prone and shoot (automatic fire). Try a few times. 

**Where did the issue occur?**
Editor (Multiplayer) 

**Placed Modules:**
 none

**RPT log file:**
not needed

**Comment by ZeusWrath** [comment]
15 days this ticket is ignored. A player cannot shoot a MX LSW machine gun! 
Someone will fix this?

**Comment by jonpas** [comment]
It is not ignored. Your report is very subjective and we have to first confirm it is actually an issue. If you want it fixed faster feel free to open a PR, otherwise this will be looked at when one of the developers (or contributors for that matter) decides to.

**Comment by nateberkopec** [comment]
A couple of things about jamming in ACE:

* Barrel temp matters. So jams after the first one will always happen faster than it took to jam the first time (unless you swapped barrels). You didn't mention barrel swapping in your test (I don't know if you can even swap the barrel in the MX LSW).
* Lying prone in dusty areas increases jam chance. You didn't mention where you were testing.

That said, see discussion at #658.

**Comment by ZeusWrath** [comment]
Hi all. I'm only 15 days waiting for any answer. Any status of this issue. But it was not... It was like in oblivion

> Barrel temp matters. So jams after the first one will always happen faster than it took to jam the first time (unless you swapped barrels). You didn't mention barrel swapping in your test (I don't know if you can even swap the barrel in the MX SW).

I haven't used the replacement barrel. It is irrelevant to this issue. The issue is that new weapon, sometimes jammed by 16-18 seconds! This is not normal. Replacing the barrel should occur after this point.

> Lying prone in dusty areas increases jam chance. You didn't mention where you were testing.

Yes, in the tests I was in a prone position, but the test passed on the Stratis airfield. On the runway. This is a paved surface, there may not be a lot of dust.  I think in any case, the weapon designed to fire long bursts, should not be jammed on 16-18 seconds  to any location on the map.

**Comment by PabstMirror** [comment]
Have a PR to reduce the rate of jams and tweak heat calcs - #4969
But also keep in mind the MX_SW is a fairly lightweight weapon compared to conventional lmgs.

**Comment by commy2** [comment]
Complain complain complain, but no solutions or initiative.

**Comment by ZeusWrath** [comment]
the fact is that if you do the tests, then you will see what ONLY the MX LSW machinegun will have the the most fast chance of jamming! Use the other machineguns, use the standard MX-rifle, all this weapons, can be are  jammed much later, than the MX LSW machine gun. Always and always, the MX LSW will be jammed much earlier than any other weapon. This problem is only one weapon

**Comment by Drofseh** [comment]
Here's some real world numbers on the M240 for comparison.

Weight: 22.2 lbs (10.34 kg)
Rate of Fire (Cyclic): 650-950 rpm
Rate of Fire (Rapid; 2-3 seconds between bursts): 200 rpm, with a barrel change every 2 minutes
Rate of Fire (Sustained; 4-5 seconds between bursts): 100 rpm, with barrel change every 10 minutes

Keep in mind that the MX LSW, at only 8.09 lbs (3.67 kg), weighs substantially less that the M240.

**Comment by nateberkopec** [comment]
Here are some more real-world numbers on mean rounds between stoppages: http://research.omicsgroup.org/index.php/M240_machine_gun#Early_history:_testing_and_adoption and http://www.sadefensejournal.com/wp/?p=164

**Comment by ZeusWrath** [comment]
> Here's some real world numbers on the M240 for comparison.
> 
> Weight: 22.2 lbs (10.34 kg)
> Rate of Fire (Cyclic): 650-950 rpm
> Rate of Fire (Rapid; 2-3 seconds between bursts): 200 rpm, with a barrel change every 2 minutes
> Rate of Fire (Sustained; 4-5 seconds between bursts): 100 rpm, with barrel change every 10 minutes
> 
> Keep in mind that the MX LSW, at only 8.09 lbs (3.67 kg), weighs substantially less than the M240.
> 
> edited for typo

This is a rather controversial data
Just watch the videos with crazy shooting-  
https://www.youtube.com/watch?v=OtFhfwrsWy4
https://www.youtube.com/watch?v=ibDD1UrNoXs

We do not see at least one jamm, until the weapon is completely destroyed.
In the case with AK, after a crazy shooting, we most likely will see a burning weapon, but not jumming! In the case of the M-16, we see a rupture of the barrel. These defects are not remediable. However, in the ACE we can see jamming at first magazine and that can be corrected/

**Comment by ZeusWrath** [comment]
Also to be considered is another issue!  If the player plays against the AI, then the AI has an undeniable advantage, because the AI does not experience the jamming weapon problem.  Thus, as seems ro me, the probability of the jammed weapon for player should be reduced to minimum or added for AI players also.

**Comment by commy2** [comment]
>Thus, as seems to me, the probability of the jammed weapon for player should be reduced to minimum or added for AI players also.
No.

**Comment by Cuel** [comment]
@ZeusWrath I'd suggst you simply turn it off then

**Comment by nateberkopec** [comment]
Tried in Zeus.

Standing, magdump: got through ~3 mags before stoppage. (~300rd)
Prone on a dirt surface: ~3.5 mags before stoppage. (~300rd)
Standing, 5 round bursts, 2 seconds between bursts: ~3.5 mags before stoppage (~300rd)

I was able to reproduce the "less than 1 mag" jam once. Weapon temp was ~2 bars (so still very green). Most times though, jams occurred at around 300 rounds (weapon temp was ~67% of max).

I think the first two numbers I obtained are okay. As @ZeusWrath so helpfully posted, most light ARs like AKs and AR-15's will simply explode after ~400-600 rounds of non-stop magdumping, so we should probably simulate a stoppage long before that, since ACE does not simulate total weapon failure.

The final one though seems wrong. I think maybe the gun should lose temperature more quickly during burst fire. I checked the temp after I jammed in the burst fire scenario and I was at ~80% of max temp, which seems inconsistent with the m240 numbers that were posted. But I think it's up for discussion.

**Comment by Drofseh** [comment]
>  I think maybe the gun should lose temperature more quickly during burst fire. 

I'd be ok with faster cooling between bursts and on spare barrels.

**Comment by commy2** [comment]
2 seconds between bursts is not enough time to cool the weapon. And of course the weapon can jam with the first mag. It can jam after every shot if you're sufficiently unlucky.

**Comment by nateberkopec** [comment]
Yeah, like I said, most of my stoppages were around 250-350 rounds. Which is pretty much fine IMO if you're magdumping with a light, AR-converted-to-squad-weapon like the MX SW.

**Comment by ZeusWrath** [comment]
> Thus, as seems to me, the probability of the jammed weapon for player should be reduced to minimum or added for AI players also.

> No.

Then such balance will give an advantage to the AI, because in a fight between an AI and a human player, the **AI will not have a jammed problem, but human player will have**  It's so easy to understand, but it looks like you did not even try to understand it

> @ZeusWrath Then I'd suggest you simply turn the feature off

Oh no, I like this feature on other weapons)

**Comment by nateberkopec** [comment]
I think the issue stems from the MX SW's weight. It basically weighs as much as an AR, so the jam code treats it like one. However, since it's got a decent cyclic, different role and a huge mag, it's way easier to jam than an AR of comparable weight. Is ACE modeling jam rate based on weight accurate? I guess IDK. Maybe it could be role-based instead.

I don't think AI weapons not jamming is some huge advantage. It takes like what, 3 seconds to clear a jam in ACE? Also, just because it's an advantage to the AI doesn't mean that's a bad thing. There are plenty of advantages for the AI in ARMA. AI have to have advantages over the players, otherwise the game would be too easy.

You could also create a mod which changes the MX SW's jam chance:

```
class CfgWeapons {
    class arifle_MX_SW_F : arifle_MX_Base_F {
        GVAR(mrbs) = 6000;
    };
}
```

**Comment by jonpas** [comment]
> It's so easy to understand, but it looks like you did not even try to understand it

Yes we do understand it and there is always room for improvement. But when it comes to AI it's hard to handle that (unless you like seeing AI standing in the open trying to unjam a weapon without looking for cover... etc - because that's out of our scope). Your best bet if you want AI to work the same is to disable it, there are no plans to add AI support for it. That being said, that also doesn't mean it won't ever happen.

**Comment by ZeusWrath** [comment]
>  But when it comes to AI it's hard to handle that (unless you like seeing AI standing in the open trying to unjam a weapon without looking for cover.

Hello, personally I do not agree with you. It will be better than not at all. Because we see AI recharging weapon the  without any cover,  however, it does not bother anyone, including you.  So why would you be embarrassed if the AI corrects the jamming in the same style?

**Comment by commy2** [comment]
It's impossible to make AI have their weapons jammed, because the method we used doesn't work for AI (`addAction` `"DefaultAction"`)

**Comment by ZeusWrath** [comment]
> It's impossible to make AI have their weapons jammed, because the method we used doesn't work for AI (addAction "DefaultAction")

Ok, I understand.  Please reduce the probability of jamming for MX LSW when the player is prone.

**Comment by commy2** [comment]
>Please reduce the probability of jamming for MX LSW when the player is prone.

No.

**Comment by PabstMirror** [comment]
There is already a PR that already tweaks these values https://github.com/acemod/ACE3/pull/4969

Let's wait until that's merged and released,
then we can adjust from there.

**Comment by bux** [comment]
Closed in #4969
